### PR TITLE
[16.0][IMP+FIX] account_reconcile_oca + account_statement_base

### PR DESCRIPTION
**Pull request by etobella**
This PR does 4  things:

- Improve search of moves on reconcile model
- Removes an error raised when accessing the statement line from the move
- Adds some fields on statement lines.
- Improves refresh of kanban view when editing the records.

### CODE REVIEW

File: account_reconcile_oca/views/account_move_line.xml [hunk 1]
```diff
@@ -40,7 +40,7 @@
                     <field
                     name="name"
                     string="Journal Item"
-                    filter_domain="[                         '|', '|', '|',                         ('name', 'ilike', self), ('ref', 'ilike', self), ('account_id', 'ilike', self), ('partner_id', 'ilike', self)]"
+                    filter_domain="['|','|','|','|','|','|',('name', 'ilike', self), ('ref', 'ilike', self), ('account_id', 'ilike', self), ('partner_id', 'ilike', self),('amount_residual', 'ilike', self),('amount_residual_currency', 'ilike', self),('discount_amount_currency', 'ilike', self)]"
```
Comment (pedrobaeza) [inline account_reconcile_oca/views/account_move_line.xml]: `discount_amount_currency` seems not very relevant

File: account_statement_base/views/account_bank_statement_line.xml [hunk 2]
```diff
@@ -72,6 +72,8 @@
                     optional="hide"
                     attrs="{'readonly': [('is_reconciled', '=', True)]}"
                 />
+            <field name="transaction_type" optional="hidden" />
```
Comment (pedrobaeza) [inline account_statement_base/views/account_bank_statement_line.xml]: Isn't this `optional="hide"`?

File: account_statement_base/views/account_bank_statement_line.xml [hunk 3]
```diff
@@ -136,6 +139,7 @@
             <field name="statement_id" />
             <field name="journal_id" domain="[('type', 'in', ('bank', 'cash'))]" />
             <field name="amount" />
+            <field name="move_id" />
```
Comment (pedrobaeza) [inline account_statement_base/views/account_bank_statement_line.xml]: Isn't this in #678?

File: account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js [hunk 4]
```diff
@@ -23,9 +23,11 @@ export class ReconcileFormController extends FormController {
     async reloadFormController() {
         var is_reconciled = this.model.root.data.is_reconciled;
         await this.model.root.load();
-        if (!is_reconciled && this.model.root.data.is_reconciled) {
-            // This only happens when we press the reconcile button
-            if (this.env.parentController) {
+        // This only happens when we press the reconcile button
```
Comment (pedrobaeza) [inline account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js]: Is this diff related to the commit message?

File: account_reconcile_oca/views/account_move_line.xml (hunk 1 above)
Comment (etobella) [reply account_reconcile_oca/views/account_move_line.xml]: Done!

File: account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js (hunk 4 above)
Comment (etobella) [reply account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js]: Yes it is. If we don't apply this change, the update is only done when  we go from not reconciled to reconciled

**Comment by etobella** [comment]
@pedrobaeza Changes applied

File: account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js (hunk 4 above)
Comment (pedrobaeza) [reply account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js]: But then the comment seems incorrect, as this happens a lot of times...

Comment (etobella) [reply account_reconcile_oca/static/src/js/reconcile_form/reconcile_form_controller.esm.js]: True, I just moved the comment to the right place (next if) and put the right comment there. Thanks

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor
